src/middlewared/middlewared/plugins/tunables.py

```diff
@@ -25,212 +26,106 @@ class TunableService(CRUDService):
+        Retrieve the supported tunable types that can be changed.
         """
         return {k: k for k in TUNABLE_TYPES}
 
     @accepts(Dict(
         'tunable_create',
-        Str('var', validators=[Match(r'^[\w\.\-]+$')], required=True),
+        Str('var', required=True),
         Str('value', required=True),
-        Str('type', enum=TUNABLE_TYPES, required=True),
+        Str('type', default='SYSCTL'),
         Str('comment'),
         Bool('enabled', default=True),
         register=True
     ))
     async def do_create(self, data):
         """
         Create a Tunable.
+        """
+        verrors = ValidationErrors()
+        if await self.middleware.call('tunable.query', [('var', '=', data['var'])]):
+            verrors.add('tunable.create', f'Tunable {data["var"]!r} already exists in database.', errno.EEXIST)
 
-        `var` represents name of the sysctl/loader/rc variable.
+        if data['var'] not in await self.middleware.call('tunable.get_system_tunables'):
+            verrors.add('tunable.create', f'Tunable {data["var"]!r} does not exist in kernel.', errno.ENOENT)
 
-        `type` for SCALE should be one of the following:
-        1) SYSCTL     -     Configure `var` for sysctl(8)
-        """
-        await self.clean(data, 'tunable_create')
-        await self.validate(data, 'tunable_create')
-        await self.lower(data)
-
-        data['id'] = await self.middleware.call(
-            'datastore.insert',
-            self._config.datastore,
-            data,
-            {'prefix': self._config.datastore_prefix}
-        )
+        if not (orig_value := (await self.middleware.call('tunable.get_or_set', data['var']))):
+            verrors.add('tunable.create', f'Unable to determine original value of {data["var"]!r}')
+
+        if data['type'] not in await self.middleware.call('tunable.tunable_type_choices'):
+            verrors.add('tunable.create', 'Invalid tunable type.')
 
-        await self.middleware.call('service.reload', data['type'])
+        verrors.check()
 
-        return await self.get_instance(data['id'])
+        data['orig_value'] = orig_value
 
-    async def do_update(self, id, data):
+        if (comment := data.get('comment', '').strip()):
+            data['comment'] = comment
+
+        _id = await self.middleware.call(
+            'datastore.insert', self._config.datastore, data, {'prefix': self._config.datastore_prefix}
+        )
+        await self.middleware.call('service.restart', 'sysctl')
+        return await self.get_instance(_id)
+
+    async def do_update(self, _id, data):
         """
         Update Tunable of `id`.
         """
-        old = await self.get_instance(id)
+        old = await self.get_instance(_id)
 
         new = old.copy()
         new.update(data)
+        if old == new:
+            # nothing updated so return early
+            return old
 
-        await self.clean(new, 'tunable_update', old=old)
-        await self.validate(new, 'tunable_update')
```

Comment:
We should also validate that if newer variable changed, we don't already have it in the database like we are doing in the creation